src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -71,6 +71,10 @@
   // added to buffer before calling insertRows
   private final AtomicLong processedOffset; // processed offset
 
+  // This offset is updated when Snowflake has received offset from insertRows API
+  // We will update this value after calling offsetToken API for this channel
+  private final AtomicLong offsetPersistedInSnowflake = new AtomicLong(-1);
```

Comment:
Why is this an `AtomicLong`?

Reply:
Umm good point. It doesnt need to be AtomicLong at the moment.

Reply:
Agreed, there is no atomic operation needed and it is single threaded at the moment.